Brick-Studios/BrickEngine
# Collision detector

# Description

Improved the collision detector so it can detect if you are in a component that has isTrigger enabled. It also returns the objectId that you are about to or are colliding with.
Changed the physics system so you can move through those components as well.

# How can this be tested?
Test with this branch: [Weapon](https://github.com/Brick-Studios/BeastArena/tree/Weapon)
Run the game and try these things:

- Run in the weapon and press E (console should output "picked up weapon")
- Run in the black square and press E (console should output "this is not a weapon")
- Press E without standing in anything (console should output "you are not in an entity")
- You should not be able to walk through gorilla

# Checklist:
